File: CollectionView/Collectionview/Collectionview/ViewController.m
//
//  ViewController.m
//  Collectionview
//
//  Created by Hayder on 16/8/4.
//  Copyright © 2016年 wangzhenhai. All rights reserved.

//  整理一个关于collectionView用法的Demo

#import "ViewController.h"
#import "header.h"

@interface ViewController ()<UITableViewDelegate,UITableViewDataSource>

@property (nonatomic, strong) NSArray *dataArr;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.title = @"CollectionView的使用";
    
    UITableView *tableview = [[UITableView alloc] initWithFrame:CGRectMake(0, 20, [UIScreen mainScreen].bounds.size.width, [UIScreen mainScreen].bounds.size.height - 20) style:UITableViewStyleGrouped];
    tableview.delegate = self;
    tableview.dataSource = self;
    [self.view addSubview:tableview];
}

#pragma mark ------------------------tableviewDelegate------------------------
- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView
{
    return self.dataArr.count;
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section
{
    return [[self.dataArr[section] objectForKey:@"data"] count];
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
{
    static NSString *ID = @"cell";
    
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:ID];
    
    if (!cell) {
        
        cell = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:ID];
    }
    
    NSArray *data = [self.dataArr[indexPath.section] objectForKey:@"data"];
    cell.textLabel.text = data[indexPath.row];
    return cell;
}

- (UIView *)tableView:(UITableView *)tableView viewForHeaderInSection:(NSInteger)section
{
    UILabel *sectionNameLabel = [[UILabel alloc] init];
    sectionNameLabel.text = [self.dataArr[section] objectForKey:@"sectionName"];
    
    return sectionNameLabel;
}

- (CGFloat)tableView:(UITableView *)tableView heightForHeaderInSection:(NSInteger)section
{
    return 30;
}

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath
{
    NSLog(@"%@",[self getCntrollerClassWithIndexPath:indexPath]);
    
    Class cls = [self getCntrollerClassWithIndexPath:indexPath];
    
    [self.navigationController pushViewController:[[cls alloc] init] animated:YES];
}

#pragma mark ------------------------lazyLoading------------------------
- (NSArray *)dataArr
{
    if (!_dataArr) {
        
        _dataArr = @[@{@"sectionName":@"基础",@"data":@[@"普通用法 ZHNormalController",@"瀑布流 ZHWaterFlowController"]},
                     @{@"sectionName":@"高级",@"data":@[]}];
    }
    return _dataArr;
}

- (Class)getCntrollerClassWithIndexPath:(NSIndexPath *)indexPath
{
    NSString *target = self.dataArr[indexPath.section][@"data"][indexPath.row];
    NSArray *components = [target componentsSeparatedByString:@" "];
    return NSClassFromString(components[1]);
}

@end

File: Demo/Swift/5.Swift3.0 - Switch/5.Swift3.0 - Switch/ViewController.swift
//
//  ViewController.swift
//  5.Swift3.0 - Switch
//
//  Created by Hayder on 16/9/22.
//  Copyright © 2016年 wangzhenhai. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    
    /**
    1.OC中 Switch() 中的值必须是整数
    2.每个语句都需要一个 break
    3.如果要定义局部变量，需要{}
    4.OC中{}可以限定变量的作用域。
     */
    override func viewDidLoad() {
        super.viewDidLoad()
        
        Demo(num: "10")
    }
    
    /**
     1.Swift 可以针对任意类型的值进行分支，不再局限整数
     2.Swift 一般不需要break
     3.Swift 如果需要多值,使用‘,’
     4.所有的分支至少需要一条指令，如果什么都不干，才使用break
     */
    func Demo(num : String){
        
        switch num {
        case "10","9":
            
            print("优")
            
        default:
            print("一般")
        }
        
    }
}

